core/src/main/java/io/parsingdata/metal/data/selection/ByToken.java

```diff
@@ -51,8 +53,29 @@ private static ParseItemList getAllRecursive(final ParseGraph graph, final Token
         final ParseItemList results = graph.definition == definition ? tailResults.add(graph) : tailResults;
         final ParseItem head = graph.head;
         if (head.isValue() && head.asValue().definition == definition) { return results.add(head); }
+        if (head.isRef() && head.asRef().definition == definition) { return results.add(head); }
```

Comment:
Agreed. But since it's a highly uncommon thing I propose we focus on the core things first :)